gobblin-service/src/test/java/org/apache/gobblin/service/modules/orchestration/OrchestratorTest.java

```diff
@@ -244,9 +250,10 @@ public void createTopologySpec() {
     }
     // Make sure TopologyCatalog is empty
     Assert.assertTrue(specs.size() == 0, "Spec store should be empty before addition");
-    // Make sure TopologyCatalog Listener is empty
-    Assert.assertTrue(specCompiler.getTopologySpecMap().size() == 0, "SpecCompiler should not know about any Topology "
-        + "before addition");
+    Assert.assertTrue(specCompiler.getTopologySpecMap().size() == 1, "SpecCompiler should know about any Topology "
+        + " irrespective of what is there in the topology catalog");
+    // Make sure TopologyCatalog empty
+    Assert.assertTrue(this.topologyCatalog.getSize() == 0, "Topology catalog should contain 0 Spec before addition");
```

Comment:
seems this asserts that the spec compiler's topos may now potentially be inconsistent w/ the topo catalog's.  that may make the system harder to reason about... is it really a good thing to drop such an invariant?

Reply:
there is a deviation only till everything is initialized. after that, they both should be same (1 in this test).

Reply:
prior to this change, there's an "inconsistent timespan", where the compiler doesn't have topo specs initialized.  you suggested this might compromise correctness of `DagManagementStateStore`.

after this change, sounds like there will be an "inconsistent timespan", where the `TopologyCatalog` may be out of sync w/ the topo specs the compiler knows about.  could there be any negative consequence to that?